client/src/components/UserMenu.js

```diff
@@ -0,0 +1,138 @@
+import React from "react";
+import {
+  Menu,
+  Button,
+  MenuItem,
+  Divider,
+  makeStyles,
+  Paper,
+  MenuList,
+  Link,
+  Popper,
+  Grow,
+  ClickAwayListener
+} from "@material-ui/core";
+import { googleLogout } from "../utils/googleAuth";
+import { useHistory } from "react-router-dom";
+import { useSetAuthenticated } from "../providers/AuthProvider";
+import ExitToAppIcon from "@material-ui/icons/ExitToApp";
+import PeopleOutlineIcon from "@material-ui/icons/PeopleOutline";
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    display: "flex"
+  },
+  paper: {
+    marginRight: theme.spacing(2)
+  },
+  link: {
+    color: "black",
+    marginRight: theme.spacing(3),
+    cursor: "pointer",
+    fontWeight: "bold"
+  },
+  Menu: {
+    marginLeft: 25
+  },
+  icon: { marginRight: 10 }
+}));
+
+export default function UserMenu({ name }) {
+  // const [anchorEl, setAnchorEl] = React.useState(null);
+
+  // const handleClick = event => {
+  //   setAnchorEl(event.currentTarget);
+  // };
+
+  // const handleClose = () => {
+  //   setAnchorEl(null);
+  // };
+
+  const classes = useStyles();
+  const history = useHistory();
+  const [open, setOpen] = React.useState(false);
+  const anchorRef = React.useRef(null);
+  const setAuthenticated = useSetAuthenticated();
+
+  const handleToggle = () => {
+    setOpen(prevOpen => !prevOpen);
```

Comment:
Would there be any difference between `setOpen(prevOpen => !prevOpen)` and `setOpen(!prevOpen)`